.github/workflows/docker-publish.yaml

```diff
@@ -0,0 +1,77 @@
+# This job will build and push the Docker image to Docker Hub.
+name: docker-publish
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        type: string
+        required: true
+        description: The release version (e.g., X.Y.Z), without the v.
+      latest:
+        type: boolean
+        description: Whether to push the image with the latest tag
+      publish:
+        type: boolean
+        description: Whether to push the image to Docker Hub
+permissions: read-all
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    if: github.repository == 'bufbuild/buf'
+    steps:
+      - name: Validate Version
+        if: ${{ startsWith(inputs.buf_version, 'v') }}
+        run: |
+          echo "error: buf version must not start with 'v'."
+          exit 1
+      - name: checkout
+        uses: actions/checkout@v4
+        with:
+          ref: refs/tags/v${{ github.event.inputs.version }}
+      - name: setup-qemu
+        uses: docker/setup-qemu-action@v3
+        id: qemu
+        with:
+          platforms: linux/386,linux/amd64,linux/arm64,linux/arm/v7,linux/arm/v6,linux/ppc64le,linux/s390x
+      - name: setup-docker-buildx
+        uses: docker/setup-buildx-action@v3
+      - name: login-docker
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_TOKEN }}
+      - name: docker-meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: |
+            bufbuild/buf
+          tags: |
+            type=raw,value=latest,enable=${{ github.event.inputs.latest }}
+            type=semver,pattern={{major}},value={{ github.event.inputs.version }}
+            type=semver,pattern={{major}}.{{minor}},value={{ github.event.inputs.version }}
+            type=semver,pattern={{major}}.{{minor}}.{{patch}},value={{ github.event.inputs.version }}
+            type=semver,pattern={{version}},value={{ github.event.inputs.version }}
```

Comment:
Does the `semver` type require the `v` prefix on values?

Reply:
Yeah looks like it does: https://github.com/docker/metadata-action?tab=readme-ov-file#typesemver  fixing.